demisto_sdk/commands/validate/validators/BC_validators/BC107_is_valid_toversion_on_modified.py

```diff
@@ -14,30 +16,112 @@
 
 class IsValidToversionOnModifiedValidator(BaseValidator[ContentTypes]):
     error_code = "BC107"
-    description = (
-        "Check that the toversion property was not changed on existing Content files."
-    )
+    description = "Checks that the `toversion` property was not changed on existing content, unless it is replaced by a new file with the same ID and a continuous fromversion."
     rationale = "Changing the `toversion` field for a content item can break backward compatibility."
-    error_message = "Changing the maximal supported version field `toversion` is not allowed. Please undo, or request a force merge."
+    error_message = "Changing the maximal supported version field `toversion` is not allowed. unless you're adding new content_item with the same id {0} and their from/to version fulfills the following:\n{1}"
     related_field = "toversion"
-    expected_git_statuses = [GitStatuses.MODIFIED, GitStatuses.RENAMED]
+    expected_git_statuses = [
+        GitStatuses.MODIFIED,
+        GitStatuses.RENAMED,
+        GitStatuses.ADDED,
+    ]
+    valid_cases = (
+        "The old item `toversion` field should be less than the new item `fromversion` field\n"
+        "The old and the new item should be continuous, aka the old one `toversion` is one version"
+        " less than the new one `fromversion`"
+    )
 
     def obtain_invalid_content_items(
         self, content_items: Iterable[ContentTypes]
     ) -> List[ValidationResult]:
+        modified_items, new_items = sort_content_items(content_items)
         return [
             ValidationResult(
                 validator=self,
-                message=self.error_message,
+                message=self.error_message.format(
+                    content_item.object_id, self.valid_cases
+                ),
                 content_object=content_item,
             )
-            for content_item in content_items
-            if toversion_modified(content_item)
+            for content_item in modified_items
+            if self.invalid_toversion(content_item, new_items)
```

Comment:
```suggestion
            if self.invalid_toversion(content_item, new_items)
```
maybe rename function name to invalid_toversion_modification or something similar?